azure-pipelines-official.yml

```diff
@@ -38,21 +38,26 @@ stages:
     - powershell: Write-Host "##vso[task.setvariable variable=SourceBranchName]$('$(Build.SourceBranch)'.Substring('refs/heads/'.Length))"
       displayName: Setting SourceBranchName variable
 
-    - powershell: Write-Host "##vso[task.setvariable variable=IsPRValidationBuild]($(PRNumber) -ne '') -and ($(PRNumber) -ne 'default')"
-      displayName: Setting IsPRValidationBuild variable
-
     - task: tagBuildOrRelease@0
-      displayName: Tag build
+      displayName: Tag official build
+      inputs:
+        type: 'Build'
+        tags: 'OfficialBuild'
+      condition: and(succeeded(), eq(variables['PRNumber'], 'default'))
+          
+    - task: tagBuildOrRelease@0
+      displayName: Tag PR validation build
       inputs:
         type: 'Build'
-        tags: if (variables['IsPRValidationBuild']) { 'PRValidationBuild' } else { "OfficialBuild" }
+        tags: 'PRValidationBuild'
+      condition: and(succeeded(), ne(variables['PRNumber'], 'default'))
 
     - task: PowerShell@2
       displayName: Setup branch for insertion validation
       inputs:
         filePath: 'eng\setup-pr-validation.ps1'
         arguments: '-sourceBranchName $(SourceBranchName) -prNumber $(PRNumber)'
```

Comment:
Has this been tested with an invalid PR number and does the pipeline fail at this point?

Reply:
This doesn't check for invalid number (with the assumption that the build should be tag as validation for any value other than 'default', including ''). The build will fail the next step